src/sinks/datadog/events.rs

```diff
@@ -0,0 +1,444 @@
+use super::healthcheck;
+use crate::{
+    config::{log_schema, DataType, GenerateConfig, SinkConfig, SinkContext, SinkDescription},
+    event::{Event, PathComponent},
+    http::HttpClient,
+    internal_events::{DatadogEventsFieldInvalid, DatadogEventsProcessed},
+    sinks::{
+        util::{
+            batch::Batch,
+            encoding::{EncodingConfigWithDefault, EncodingConfiguration, TimestampFormat},
+            http::{HttpSink, PartitionHttpSink},
+            BatchConfig, BatchSettings, BoxedRawValue, Concurrency, EncodedEvent, JsonArrayBuffer,
+            PartitionBuffer, PartitionInnerBuffer, TowerRequestConfig,
+        },
+        Healthcheck, VectorSink,
+    },
+    tls::{MaybeTlsSettings, TlsConfig},
+};
+use futures::{FutureExt, SinkExt};
+use http::Request;
+use indoc::indoc;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::{sync::Arc, time::Duration};
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct DatadogEventsConfig {
+    endpoint: Option<String>,
+
+    #[serde(default = "default_site")]
+    site: String,
+    default_api_key: String,
+
+    tls: Option<TlsConfig>,
+
+    #[serde(default)]
+    request: TowerRequestConfig<Concurrency>,
+}
+
+type ApiKey = Arc<str>;
+
+fn default_site() -> String {
+    "datadoghq.com".to_owned()
+}
+
+inventory::submit! {
+    SinkDescription::new::<DatadogEventsConfig>("datadog_events")
+}
+
+impl GenerateConfig for DatadogEventsConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(indoc! {r#"
+            default_api_key = "${DATADOG_API_KEY_ENV_VAR}"
+        "#})
+        .unwrap()
```

Comment:
Could this be done with `toml::Value::try_from(Self { … })` or even the `impl_generate_config_from_default` macro?

Reply:
It can but then we can't show `${DATADOG_API_KEY_ENV_VAR}`. Which is not necessary but it's neat.